src/management/users.ts

```diff
@@ -79,7 +79,7 @@ export default class Users {
    *
    * @memberof Users
    */
-  patchUser(parameters: PatchUserOptions) {
+  patchUser(parameters: PatchUserOptions): Promise<Users> {
```

Comment:
This is not returning the correct type. Users refers to the ManagementClient itself, while I guess you want User.
Again, this is where TypeScript should help you, but it's not because all of the any's and unknowns.